src/main/java/de/abstractolotl/azplace/rest/controller/AuthController.java

```diff
@@ -24,42 +26,83 @@
 import org.springframework.web.server.ResponseStatusException;
 
 import javax.servlet.http.HttpSession;
+import java.util.Objects;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-@RestController
+@RestController @Slf4j
 public class AuthController implements AuthAPI {
 
     private final static JsonMapper jsonMapper = new JsonMapper();
 
-    @Value("${app.cas.redirecturl}") private String redirectUrl;
-    @Value("${app.defaultKeyValidTime}") private int defaultKeyValidTime;
-    @Value("${app.cas.url}") private String casUrl;
-    @Value("${app.cas.apiurl}") private String apiUrl;
+    @Value("${oauth.redirectUrl}")
+    private String redirectUrl;
+    @Value("${oauth.authorizationEndpoint}")
+    private String authorizationEndpoint;
+    @Value("${oauth.tokenEndpoint}")
+    private String tokenEndpoint;
+    @Value("${oauth.userInfoEndpoint}")
+    private String userInfoEndpoint;
 
-    @Autowired private AuthenticationService authService;
-    @Autowired private ElasticService elasticService;
+    @Value("${oauth.clientId}")
+    private String clientId;
+    @Value("${oauth.clientSecret}")
+    private String clientSecret;
 
-    @Autowired private UserRepo userRepo;
+    @Value("${oauth.frontendUrl}")
+    private String frontendUrl;
 
-    @Autowired private UserSession session;
+    private final AuthenticationService authService;
+    private final ElasticService elasticService;
+
+    private final UserRepo userRepo;
+
+    private final UserSession session;
+
+    @Autowired
+    public AuthController(AuthenticationService authService, ElasticService elasticService, UserRepo userRepo, UserSession session) {
+        this.authService = authService;
+        this.elasticService = elasticService;
+        this.userRepo = userRepo;
+        this.session = session;
+    }
 
     @Override
     public String login(String hostName) {
-        String url = casUrl + "/login?service=https://" + hostName + "/auth/verify";
+        String url = authorizationEndpoint + "?response_type=code&client_id=" + clientId + "&redirect_uri=" + redirectUrl + "&scope=openid profile";
         return "<meta http-equiv=\"refresh\" content=\"0; url=" + url + "\" />";
     }
 
     @Override
-    public ResponseEntity<String> verify(String ticket) {
-        CASUser casUser = validateTicket(ticket);
-        User user = createOrGetUser(casUser);
+    public ResponseEntity<String> verify(String code) {
+        String validateCode = validateCode(code);
 
-        session.setUser(user);
-        elasticService.logLogin();
+        try {
+            JsonNode json = jsonMapper.readValue(validateCode, ObjectNode.class);
```

Comment:
Why parse username, first name, etc manual?
Would be better to create a class with the needed attibutes and parse it to that, instead of ObjectNode